controllers/apps/components/internal/component_base_stateful_hscale_backup.go

```diff
@@ -0,0 +1,676 @@
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+type backupDataClone struct {
+	baseDataClone
+}
+
+var _ DataClone = &backupDataClone{}
+
+func (d *backupDataClone) Succeed() (bool, error) {
+	allPVCsExist, err := d.checkAllPVCsExist()
+	if err != nil || !allPVCsExist {
+		return allPVCsExist, err
+	}
+	pvcKeys := d.toCreatePVCKeys()
+	for _, pvcKey := range pvcKeys {
+		restoreStatus, err := d.checkRestoreStatus(pvcKey)
+		if err != nil {
+			return false, err
+		}
+		if restoreStatus != backupStatusReadyToUse {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (d *backupDataClone) ClearTmpResources() ([]client.Object, error) {
+	objs := make([]client.Object, 0)
+	// delete backup
+	ml := d.getBackupMatchingLabels()
+	backupList := dataprotectionv1alpha1.BackupList{}
+	if err := d.cli.List(d.reqCtx.Ctx, &backupList, ml); err != nil {
+		return nil, err
+	}
+	for i := range backupList.Items {
+		objs = append(objs, &backupList.Items[i])
+	}
+	// delete restore job
+	jobList := v1.JobList{}
+	if err := d.cli.List(d.reqCtx.Ctx, &jobList, ml); err != nil {
+		return nil, err
+	}
+	for i := range jobList.Items {
+		objs = append(objs, &jobList.Items[i])
+	}
+	return objs, nil
+}
+
+func (d *backupDataClone) backup() ([]client.Object, error) {
+	objs := make([]client.Object, 0)
+	backupPolicyTplName := d.component.HorizontalScalePolicy.BackupPolicyTemplateName
+	backupPolicy, err := GetBackupPolicyFromTemplate(d.reqCtx, d.cli, d.cluster, d.component.ComponentDef, backupPolicyTplName)
+	if err != nil {
+		return nil, err
+	}
+	if backupPolicy == nil {
+		return nil, intctrlutil.NewNotFound("not found any backup policy created by %s", backupPolicyTplName)
+	}
+	backup, err := builder.BuildBackup(d.cluster, d.component, backupPolicy.Name, d.key, "datafile")
+	if err != nil {
+		return nil, err
+	}
+	objs = append(objs, backup)
+	return objs, nil
+}
+
+func (d *backupDataClone) checkBackupStatus() (backupStatus, error) {
+	backup := dataprotectionv1alpha1.Backup{}
+	if err := d.cli.Get(d.reqCtx.Ctx, d.key, &backup); err != nil {
+		if errors.IsNotFound(err) {
+			return backupStatusNotCreated, nil
+		} else {
+			return backupStatusFailed, err
+		}
+	}
+	if backup.Status.Phase == dataprotectionv1alpha1.BackupFailed {
+		return backupStatusFailed, fmt.Errorf("failed to backup: %s", backup.Status.FailureReason)
+	}
+	if backup.Status.Phase == dataprotectionv1alpha1.BackupCompleted {
+		return backupStatusReadyToUse, nil
+	}
+	return backupStatusProcessing, nil
+}
+
+func (d *backupDataClone) restore(pvcKey types.NamespacedName) ([]client.Object, error) {
+	objs := make([]client.Object, 0)
+	restoreJobKey := d.restoreKeyFromPVCKey(pvcKey)
+	backup := dataprotectionv1alpha1.Backup{}
+	if err := d.cli.Get(d.reqCtx.Ctx, d.key, &backup); err != nil {
+		return nil, err
+	}
+	ml := client.MatchingLabels{
+		constant.ClusterDefLabelKey: d.component.ClusterDefName,
+	}
+	backupToolList := dataprotectionv1alpha1.BackupToolList{}
```

Comment:
you can get backupToolName from backup CR via`backup.Status.BackupToolName`

Reply:
also need backuptool image as job container image to create restore job